lms/djangoapps/discussion_api/views.py

```diff
@@ -75,27 +76,41 @@ class CourseTopicsView(DeveloperErrorViewMixin, APIView):
     **Example Requests**:
 
         GET /api/discussion/v1/course_topics/course-v1:ExampleX+Subject101+2015
+            ?topic_id=a29d117faf43c7c59eda7717e722e9d8f4a47ea8
 
     **Response Values**:
+        if topic_id is not given:
+            * courseware_topics: The list of topic trees for courseware-linked
+              topics. Each item in the list includes:
 
-        * courseware_topics: The list of topic trees for courseware-linked
-          topics. Each item in the list includes:
+                * id: The id of the discussion topic (null for a topic that only
+                  has children but cannot contain threads itself).
 
-            * id: The id of the discussion topic (null for a topic that only
-              has children but cannot contain threads itself).
+                * name: The display name of the topic.
+
+                * children: A list of child subtrees of the same format.
+
+            * non_courseware_topics: The list of topic trees that are not linked to
+                  courseware. Items are of the same format as in courseware_topics.
+
+        if topic_id is given:
+            * id: The id of the discussion topic.
 
             * name: The display name of the topic.
 
             * children: A list of child subtrees of the same format.
-
-        * non_courseware_topics: The list of topic trees that are not linked to
-              courseware. Items are of the same format as in courseware_topics.
     """
```

Comment:
@robrap I could be persuaded either way.  topic is already mentioned in the `/course_topics/`segment of the URL, and I tend to like URLs that make sense if you take away segments from the end, and 

    /api/discussion/v1/course_topics/course-v1:ExampleX+Subject101+2015/topic/

doesn't represent anything real.  On the other hand, 

    /api/discussion/v1/course_topics/course-v1:ExampleX+Subject101+2015/a29d117faf43c7c59eda7717e722e9d8f4a47ea8

is pretty hard to decipher.

I don't think edX has an established goal of making all URL segments meaningful, though, so unless someone else has a strong opinion, let's go with your first example.